src/Polymorph-Widgets/ThemeIcons.class.st

```diff
@@ -300,20 +301,20 @@ ThemeIcons >> iconNamed: aSymbol [
 			self isReportingNotFound
 				ifTrue: [
 					self crTrace: (aSymbol, ' icon not found!').
-					self notFoundIcon ]
-				ifFalse: [ self blankIcon ]]) scaledByDisplayScaleFactor
+					self notFoundIcon scaledByDisplayScaleFactor ]
+				ifFalse: [ self blankIcon scaledByDisplayScaleFactor ]])
 ]
 
 { #category : #accessing }
 ThemeIcons >> iconNamed: aSymbol ifNone: aBlock [
 
-	self icons at: aSymbol asSymbol ifPresent: [ :icon | ^ icon ].
+	self icons at: aSymbol asSymbol ifPresent: [ :icon | ^ icon scaledToSize: icon extent * (self currentWorld displayScaleFactor / scale) ].
 	"Trying the old way"
 
 	((aSymbol endsWith: 'Icon') or: [ (aSymbol endsWith: 'Form') ]) ifTrue: [
```

Comment:
Not for this PR: but we really need to find a way to get rid of the "old way", to force people to change the names. I did a session to reduce #asSymbol sends during normal use, this happend *a  lot* that this code path was taken. I checked again, and, oh wonder, *new code* was written that end here agan, for every time we open a menu. 

Why even change to a "new way" if there is no way to slowly refactor away from the "old way"?

(not for this PR)

Reply:
We should add more rules :)